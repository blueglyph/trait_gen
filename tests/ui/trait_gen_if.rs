#![allow(unused)]
use trait_gen::{trait_gen, trait_gen_if};

trait Binary {
    const MSB: u32;
}

// missing generic argument
#[trait_gen(T -> u8, u16, u32)]
impl Binary for U {
    #[trait_gen_if( in u8)]
    const MSB: u32 = 7;
}

// '->' instead of 'in'
#[trait_gen(T -> u8, u16, u32)]
impl Binary for U {
    #[trait_gen_if(U -> u8)]
    const MSB: u32 = 7;
}

// missing types
#[trait_gen(T -> u8, u16, u32)]
impl Binary for U {
    #[trait_gen_if(T in)]
    const MSB: u32 = 7;
}

// missing punctuation
#[trait_gen(T -> u8, &u8, u16, &u16)]
impl Binary for U {
    #[trait_gen_if(T in u8 &u8)]
    const MSB: u32 = 7;
}

// missing arguments
#[trait_gen(T -> u8, &u8, u16, &u16)]
impl Binary for U {
    #[trait_gen_if]
    const MSB: u32 = 7;
}

// missing arguments
#[trait_gen(T -> u8, &u8, u16, &u16)]
impl Binary for U {
    #[trait_gen_if()]
    const MSB: u32 = 7;
}

fn main() {}