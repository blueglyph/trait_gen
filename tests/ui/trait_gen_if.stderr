error: expected substitution identifier or type

         = help: The expected format is: #[trait_gen_if(T in u8, u16, u32)]

  --> tests/ui/trait_gen_if.rs:11:21
   |
11 |     #[trait_gen_if( in u8)]
   |                     ^^

error: expected `in`

         = help: The expected format is: #[trait_gen_if(T in u8, u16, u32)]

  --> tests/ui/trait_gen_if.rs:18:22
   |
18 |     #[trait_gen_if(U -> u8)]
   |                      ^

error: expected type after 'in'

         = help: The expected format is: #[trait_gen_if(T in u8, u16, u32)]

  --> tests/ui/trait_gen_if.rs:25:24
   |
25 |     #[trait_gen_if(T in)]
   |                        ^

error: expected `,`

         = help: The expected format is: #[trait_gen_if(T in u8, &u8, u16)]

  --> tests/ui/trait_gen_if.rs:32:28
   |
32 |     #[trait_gen_if(T in u8 &u8)]
   |                            ^

error: expected attribute arguments in parentheses: #[trait_gen_if(...)]

         = help: The expected format is: #[trait_gen_if(T in u8, &u8, u16)]

  --> tests/ui/trait_gen_if.rs:39:7
   |
39 |     #[trait_gen_if]
   |       ^^^^^^^^^^^^

error: expected substitution identifier or type

         = help: The expected format is: #[trait_gen_if(T in u8, &u8, u16)]

  --> tests/ui/trait_gen_if.rs:46:20
   |
46 |     #[trait_gen_if()]
   |                    ^
